### PR TITLE
Process URL that have a page param

### PR DESCRIPTION
**Pull request by dwhenry**
We have added this to allow the crawler to follow through multiple pages in the
search results, this should help ensure that we archive every page on the site.

We ensure that URL's with a page command are crawled, however we do not write
them to disk.

https://trello.com/c/5Le8O6cb/120-crawl-pages-with-page-param

### CODE REVIEW

File: workflow.go [hunk 1]
```diff
@@ -258,13 +262,21 @@ func ExtractURLs(extractChannel <-chan *CrawlerMessageItem) (<-chan *url.URL, <-
...
-		if(url.RawQuery != "") {
-			log.Debugln("Skipping URL as it has query params:", u)
-			continue
+		if(uu.RawQuery != "") {
+			values, err := url.ParseQuery(uu.RawQuery)
+			if err != nil {
+				log.Debugln("Skipping URL unable to decode query params:", u)
+				continue
+			}
+			values.Del("path")
```
Comment (timblair) [inline workflow.go]: Should this be `page`, not `path`? Either way, I'd say it's worth a comment to say why we're removing this apparently magic value.

File: crawler_message_item.go [hunk 2]
```diff
@@ -34,6 +34,19 @@ func (c *CrawlerMessageItem) URL() string {
 	return string(c.Body)
 }
 
+func (c *CrawlerMessageItem) hasParams() bool {
+	urlParts, err := url.Parse(c.URL())
+	if err != nil {
```
Comment (timblair) [inline crawler_message_item.go]: You could shorten this to just a single return statement:

```go
return (err != nil || urlParts.RawQuery != "")
```

Comment (dwhenry) [reply crawler_message_item.go]: good spot

File: workflow.go (hunk 1 above)
Comment (dwhenry) [reply workflow.go]: Yes, that is correct. have updated
